### PR TITLE
RavenDB-17243: Support for empty at the index entry writer level.

### PR DESCRIPTION
**Pull request by redknightlois**

### CODE REVIEW

**Comment by maciejaszyk** [comment]
The RavenDB part of this PR is missing.

**Comment by redknightlois** [comment]
What you mean? What I am missing?

File: src/Raven.Server/Documents/Indexes/Persistence/Corax/CoraxDocumentConverterBase.cs [hunk 1]
```diff
@@ -304,7 +304,7 @@ public IndexFieldsMapping GetKnownFields()
                 //https://issues.hibernatingrhinos.com/issue/RavenDB-18146
                 throw new NotSupportedException("Boosting in index is not supported by Corax. You can do it during querying or change index type into Lucene.");
             case ValueType.EmptyString:
-                scope.Write(field.Id, EmptyString.Span, ref entryWriter);
+                scope.Write(field.Id, ReadOnlySpan<byte>.Empty, ref entryWriter);
```
Comment (redknightlois) [inline src/Raven.Server/Documents/Indexes/Persistence/Corax/CoraxDocumentConverterBase.cs:307]: @maciejaszyk Found it.

**Comment by arekpalinski** [comment]
test this please

**Comment by arekpalinski** [comment]
On macOS we got (https://github.com/ravendb/ravendb/runs/6628780917?check_suite_focus=true):

```
Failed FastTests.Corax.PrimitivesTests.OutputCompatibilityForVectorizedSingleRun(seed: 125, ai: 1, bi: 24, ss: 1) [< 1 ms]
  Error Message:
   System.PlatformNotSupportedException : Operation is not supported on this platform.
  Stack Trace:
     at System.Runtime.Intrinsics.X86.Avx2.CompareEqual(Vector256`1 left, Vector256`1 right)
   at Corax.Queries.MergeHelper.AndVectorized(Span`1 dst, Span`1 left, Span`1 right) in /Users/runner/work/ravendb/ravendb/src/Corax/Queries/MergeHelper.cs:line 145
   at FastTests.Corax.PrimitivesTests.OutputCompatibilityForVectorizedSingleRun(Int32 seed, Int32 ai, Int32 bi, Int32 ss) in /Users/runner/work/ravendb/ravendb/test/FastTests/Corax/Primitives.cs:line 141
   ```

**Comment by arekpalinski** [comment]
Also here we got failure in a test with `AndIn` (https://github.com/ravendb/ravendb/runs/6628781130?check_suite_focus=true):

```
Failed FastTests.Corax.IndexSearcherTest.AndInStatementAndWhitespaceTokenizer(setSize: 100000, stackSize: 2046) [5 s]
  Error Message:
   Assert.Equal() Failure
Expected: 33333
Actual:   33306
  Stack Trace:
     at FastTests.Corax.IndexSearcherTest.AndInStatementAndWhitespaceTokenizer(Int32 setSize, Int32 stackSize) in D:\a\ravendb\ravendb\test\FastTests\Corax\IndexSearcher.cs:line 1470

**Comment by arekpalinski** [comment]
test this please

**Comment by arekpalinski** [comment]
Also please check `SlowTests.Issues.Q14235692.SlowTests.Issues.Q14235692.Empty_Strings_Can_Be_Used_In_Where_In_With_Other_Data(options: DatabaseMode = Single , SearchEngineMode = Corax)` 

and

`[SlowTests.Bugs.Queries.QueryingOnEmptyString.SlowTests.Bugs.Queries.QueryingOnEmptyString.CanFindByemptyStringMatch(options: DatabaseMode = Single , SearchEngineMode = Corax)](https://jenkins.hibernatingrhinos.com/job/PR_Tests/2046/testReport/junit/SlowTests.Bugs.Queries/QueryingOnEmptyString/SlowTests_Bugs_Queries_QueryingOnEmptyString_CanFindByemptyStringMatch_options___DatabaseMode___Single___SearchEngineMode___Corax_/)` 
@maciejaszyk can you take a look as well?

**Comment by redknightlois** [comment]
@arekpalinski Is this the first time you see these MacOS error and in `AndIn`?

EDIT: Just pushed some more stringent initialization, to see how it behaves on the test server.

**Pull request merged**